python-frontend/src/main/java/org/sonar/python/tree/DynamicObjectInfoStatementImpl.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.tree;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.sonar.plugins.python.api.tree.DynamicObjectInfoStatement;
+import org.sonar.plugins.python.api.tree.Token;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.plugins.python.api.tree.TreeVisitor;
+
+public class DynamicObjectInfoStatementImpl extends PyTree implements DynamicObjectInfoStatement {
+
+
+  private final List<Token> questionMarksBefore;
+  private final List<Tree> children;
+  private final List<Token> questionMarksAfter;
```

Comment:
Since we won't do anything with those question marks for now, and we don't provide API to access them, I think it's okay not to differentiate them from the other tokens and just to maintain a `List<Token>` here. This would also make the creation of the object much simpler from the `IPythonTreeMaker` (just collect and add all tokens).